addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1310,6 +1310,79 @@ registry.many2one = SnippetOption.extend({
     }
 });
 
+/**
+ * Handles the edition of snippet's anchor name.
+ */
+registry.anchorName = SnippetOption.extend({
+    xmlDependencies: ['/web_editor/static/src/xml/editor.xml'],
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @public
+     */
+    onClone: function () {
+        this.$target.removeAttr('id');
+        this.$target.trigger('content_changed');
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    openAnchorDialog: function (previewMode, value, $opt) {
+        var self = this;
+        this.anchorDialog = new Dialog(this, {
+            title: _t("Choose a name for your anchor"),
+            $content: $(qweb.render('web_editor.dialog.anchorName', {
+                currentAnchor: self.$target.attr('id'),
+            })),
+            buttons: [
+                {
+                    text: _t("Save"),
+                    classes: 'btn-primary',
+                    click: function () {
+                        var $input = this.$('.o_input_anchor_name');
+                        var anchorName = $input.val().trim().split(' ').join('_');
+                        var alreadyExists = $('#' + anchorName).length > 0;
+                        $input.toggleClass('is-invalid', !alreadyExists);
+                        if (!alreadyExists) {
+                            self._setAnchorName(anchorName);
+                        } else {
+                            $input.toggleClass('is-invalid');
```

Comment:
Two `toggleClass` of the same class on the same element with opposite condition...